pkg/cortex/modules.go

```diff
@@ -765,11 +770,36 @@ func (t *Cortex) initQueryScheduler() (services.Service, error) {
 	return s, nil
 }
 
+func (t *Cortex) initResourceMonitor() (services.Service, error) {
+	if t.Cfg.ResourceThresholds.CPU <= 0 && t.Cfg.ResourceThresholds.Heap <= 0 {
```

Comment:
My vision for resource monitor is that we can use it more than just throttling but also job scheduling, dynamic query concurrency, etc.

Let me know if this makes sense.
- We have separate flag to enable monitoring/scanning for CPU and Mem. Maybe we can only monitor CPU or Mem for different components but for now we can have 1 flag to control both.
- Thresholds for CPU and Mem are independent from monitor. Even if we don't set a threshold we can still have resource monitor/scanner.